frontend/src/concepts/modelRegistry/apiHooks/useModelRegistries.ts

```diff
@@ -1,11 +1,65 @@
 import React from 'react';
-import useFetchState, { FetchState } from '~/utilities/useFetchState';
-import { ModelRegistryKind } from '~/k8sTypes';
-import { listModelRegistries } from '~/api';
+import useFetchState, { FetchState, FetchStateCallbackPromise } from '~/utilities/useFetchState';
+import { AccessReviewResourceAttributes, ServiceKind } from '~/k8sTypes';
+import {
+  ServiceModel,
+  useAccessReview,
+  useRulesReview,
+  useServicesByNames,
+  useServices,
+} from '~/api';
+import { MODEL_REGISTRY_DEFAULT_NAMESPACE } from '~/concepts/modelRegistry/const';
 
-const useModelRegistries = (): FetchState<ModelRegistryKind[]> => {
-  const getModelRegistries = React.useCallback(() => listModelRegistries(), []);
-  return useFetchState<ModelRegistryKind[]>(getModelRegistries, []);
+const accessReviewResource: AccessReviewResourceAttributes = {
+  group: 'user.openshift.io',
+  resource: 'services',
+  verb: 'list',
+  namespace: MODEL_REGISTRY_DEFAULT_NAMESPACE,
 };
 
-export default useModelRegistries;
+const useModelRegistryServices = (): FetchState<ServiceKind[]> => {
+  const [allowList, accessReviewLoaded] = useAccessReview(accessReviewResource);
+  const [loaded, statuses] = useRulesReview(MODEL_REGISTRY_DEFAULT_NAMESPACE);
+  const serviceNames = React.useMemo(() => {
+    if (!loaded) {
+      return [];
+    }
+    return (
+      statuses?.resourceRules.find(
+        ({ resources, verbs }) =>
+          resources?.includes(ServiceModel.plural) && verbs.some((verb) => verb === 'get'),
+      )?.resourceNames || []
+    );
+  }, [loaded, statuses]);
+
+  const [nonAdminServices, nonAdminServicesLoaded] = useServicesByNames(
+    serviceNames,
+    MODEL_REGISTRY_DEFAULT_NAMESPACE,
+  );
+
+  const [adminServices, adminServicesLoaded] = useServices(MODEL_REGISTRY_DEFAULT_NAMESPACE);
+
+  const getModelRegistryServices = React.useCallback<
+    FetchStateCallbackPromise<ServiceKind[]>
+  >(async () => {
+    if (!accessReviewLoaded || (!nonAdminServicesLoaded && !adminServicesLoaded)) {
+      return [];
+    }
+
+    const services = allowList ? adminServices : nonAdminServices;
+    return Promise.resolve(services);
+  }, [
+    accessReviewLoaded,
+    nonAdminServicesLoaded,
+    adminServicesLoaded,
+    allowList,
+    adminServices,
+    nonAdminServices,
+  ]);
+
+  return useFetchState<ServiceKind[]>(getModelRegistryServices, [], {
+    initialPromisePurity: true,
+  });
+};
+
+export default useModelRegistryServices;
```

Comment:
We should rename the file `useModelRegistryServices.ts` and make sure its imports are using it by the new name as well (it's still called `useModelRegistries` in `ModelRegistrySelectorContext`).